main.go

```diff
@@ -106,11 +108,62 @@ func runGateway(c *cli.Context, n *server.Node) error {
 	var ch csp.Chan                         // TODO:  csp.NewChan()
 	n.Vat.Export(chanCap, chanProvider{ch}) // TODO: Will this also be exported to peers we find in the future?
 
-	return errors.New("NOT IMPLEMENTED")
+	logger.Info("starting server, listening on port " + getServerPort())
+
+	http.HandleFunc("/echo", EchoHandler)
+	http.HandleFunc("/slight-echo", SlightEchoHandler)
+
+	return http.ListenAndServe(":8080", nil) // Can test with: curl -X GET -H "Content-Type: application/json" -d '{"message": "Hello, World!"}' http://localhost:8080/slight-echo
+}
+
+// EchoHandler echos back the request as a response
+// From https://github.com/aautar/go-http-echo/blob/master/echo.go#L21-L40
+func EchoHandler(writer http.ResponseWriter, request *http.Request) {
+
+	logger.Info("Echoing back request made to " + request.URL.Path + " to client (" + request.RemoteAddr + ")")
+
+	writer.Header().Set("Access-Control-Allow-Origin", "*")
```

Comment:
Setting headers is unnecessary.  I'd remove this from both handlers, just to keep the visual noise down.